src/pybind/mgr/dashboard/controllers/nfs.py

```diff
@@ -145,11 +145,11 @@ def create(self, path, cluster_id, pseudo, access_type,
             'fsal': fsal,
             'clients': clients
         }
-        ret, _, err = export_mgr.apply_export(cluster_id, json.dumps(raw_ex))
-        if ret == 0:
+        applied_exports = export_mgr.apply_export(cluster_id, json.dumps(raw_ex))
```

Comment:
I agree. I am not fond of using remote calls either. But its being done all over for the nfs exports call and I am not sure how much of it I can replace with the CLI. So maybe we could try exploring the other suggestions though. But that would/should be done on all the areas where the `remote` is being implemented?

Also w.r.t to testing, we don't have a valid unit tests there but the Integration tests (kcli cephadm)caught this. Cypress was erroring  while creating nfs exports because the backend is throwing out this exception. Since kcli is unstable for a long time, this was missed. 

Part of my intention for this fix is to stabilize the kcli as well, so that we can again (or start) relying on it..

Reply:
Ah, it's great then that kcli CI caught this! I was afraid that this went unnoticed.

Regarding the firs topic, what's more impactful (and less demanding) is to make everyone else (contributors to other modules) aware of who's using what, so what about adding a dummy decorator like:
```py
  @remote_call(by=['dashboard'])
  def apply_export(self, ...): 
```

A comment alone should work either:
```py
  # This method is used by the dashboard module (../dashboard/controllers/nfs.py)
  # Do not change interface without updating the Dashboard code
  def apply_export(self, ...): 
```

Mypy won't ever catch this 'cause the method is used through a CPython wrapper which changes the signature. The only way to avoid that would be to add a new `mgr.safe_remote()` with a predefined/fixed API (e.g.: a fixed return type).